src/wallet/coinselection.cpp

```diff
@@ -3,12 +3,13 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include <wallet/coinselection.h>
+#include <wallet/wallet.h>
```

Comment:
Can you avoid this cyclic module dependency (between coinselection and wallet) by moving `OutputGroup` to coinselection.h?

Reply:
`OutputGroup` depends heavily on `COutput`, which is integral to the wallet. I could move it, but it feels like `COutput` doesn't really belong in coin selection.

Reply:
Why not? See the wallet as a user of the coin selection algorithm, filling in the relevant data in its native data type.

Reply:
Oh, I see. COutput contains a CWalletTx. Yeah, no, that doesn't belong in coinselection.

But why does the coin selection code need access to CWalletTx's?

Reply:
It needs the `tx->tx->vout[output.i].nValue` and `tx->IsEquivalentTo()` (but can probably work around the latter).

I could replace this with a new struct without the `CWalletTx` reference, if I provided the values, txid, and index, I think.

Reply:
Actually, I can probably use `CInputCoin` if I keep the value ivar. Will try that.

Reply:
`CInputCoin` worked great after some tweakery.